src/esp32c3/src/flash.c

```diff
@@ -4,16 +4,85 @@
  * SPDX-License-Identifier: Apache-2.0 OR MIT
  */
 
-#include "flash.h"
+#include <target/flash.h>
+
+#include <stdbool.h>
+
+#include <log.h>
+#include <private/rom.h>
+#include <private/soc_utils.h>
+
+#include "soc/spi_mem_reg.h"
+
+/* ROM */
+
+extern uint32_t ets_efuse_get_spiconfig(void);
+extern void esp_rom_spi_flash_update_id(void);
+extern void esp_rom_spiflash_attach(uint32_t ishspi, bool legacy);
+extern int esp_rom_spiflash_unlock(void);
+
+typedef struct {
+    esp_rom_spiflash_chip_t chip;
+    uint8_t dummy_len_plus[3];
+    uint8_t sig_matrix;
+} esp_rom_spiflash_legacy_data_t;
```

Comment:
General suggestion; Please use space between codes which have different purpose.  It will be easier to read

For example
```
typedefs

extern

macros

static 

```